src/winforms/toga_winforms/window.py

```diff
@@ -5,10 +5,10 @@
 
 
 class WinFormsViewport:
-    def __init__(self, native, frame):
+    def __init__(self, native, frame, dpi):
         self.native = native
         self.frame = frame
-        self.dpi = self.native.CreateGraphics().DpiX
```

Comment:
This is a variant on a theme of the Cocoa viewport. In this case, the DPI actually *can* change; so it' makes sense to have a single source of truth, rather than a value that can be cached (and thus, potentially, be a stale cache). 

I'd rather see the viewport accessing the source of truth (the window impl), rather than caching a value. That way, if there *is* a change in DPI, we don't need to hunt down all the stale caches.